src/plopp/widgets/clip3d.py

```diff
@@ -0,0 +1,397 @@
+# SPDX-License-Identifier: BSD-3-Clause
+# Copyright (c) 2024 Scipp contributors (https://github.com/scipp)
+
+import uuid
+from functools import partial, reduce
+from typing import Any, Callable, Dict, List, Literal, Tuple
+
+import ipywidgets as ipw
+import numpy as np
+import pythreejs as p3
+import scipp as sc
+
+from ..core import Node
+from ..graphics import BaseFig
+from .debounce import debounce
+from .style import BUTTON_LAYOUT
+
+
+def _xor(x: List[sc.Variable]) -> sc.Variable:
+    dim = uuid.uuid4().hex
+    return sc.concat(x, dim).sum(dim) == sc.scalar(1, unit=None)
+
+
+OPERATIONS = {
+    'or': partial(reduce, lambda x, y: sc.logical_or(x, y)),
+    'and': partial(reduce, lambda x, y: sc.logical_and(x, y)),
+    'xor': _xor,
+}
+
+
+def select(da: sc.DataArray, s: Tuple[str, sc.Variable]) -> sc.DataArray:
+    return da[s]
+
+
+class Clip3dTool(ipw.HBox):
+    """
+    A tool that provides a slider to extract a slab of points in a three-dimensional
+    scatter plot, and add it to the scene as an opaque cut. The slider controls the
+    position and range of the slice. When the slider is dragged, the red outline of the
+    cut is moved at the same time, while the actual point cloud gets updated less
+    frequently using a debounce mechanism.
+
+    .. versionadded:: 24.04.0
+
+    Parameters
+    ----------
+    limits:
+        The spatial extent of the points in the 3d figure in the XYZ directions.
+    direction:
+        The direction normal to the slice.
+    plain_update:
```

Comment:
What's the function signature of the update function? Unless it's something that's already written elsewhere that should probably be part of the docs here.